### ISSUE TITLE
format_tool_to_openai_function includes title and description, when it is not necessary 

### ISSUE THREAD

**Issue by ahakanbaba** [langchain-ai/langchain#6324]
### System Info

Langchain version langchain==0.0.201 

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [X] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

I understand the `format_tool_to_openai_function` function was added very recently. Thank you very much for that. 

As an example for this ticket: When `format_tool_to_openai_function` is used like this 

```
from langchain.tools import FileSearchTool
from langchain.tools import format_tool_to_openai_function

format_tool_to_openai_function(FileSearchTool())
```

On a FileSearchTool it generates this 

```
{'name': 'file_search',
  'description': 'Recursively search for files in a subdirectory that match the regex pattern',
  'parameters': {'title': 'FileSearchInput',
   'description': 'Input for FileSearchTool.',
   'type': 'object',
   'properties': {'dir_path': {'title': 'Dir Path',
     'description': 'Subdirectory to search in.',
     'default': '.',
     'type': 'string'},
    'pattern': {'title': 'Pattern',
     'description': 'Unix shell regex, where * matches everything.',
     'type': 'string'}},
   'required': ['pattern']}}
```

This includes some unnecessary fields like `title` and `description` for the `parameters` and `title` for the `parameters.properties`

I do not know whether this is a correctness issue yet with the OpenAI functions api. But at least this should cause wasted tokens. 

### Expected behavior

```
{'name': 'file_search',
  'description': 'Recursively search for files in a subdirectory that match the regex pattern',
  'parameters': {
   'type': 'object',
   'properties': {'dir_path': {
     'description': 'Subdirectory to search in.',
     'default': '.',
     'type': 'string'},
    'pattern': {
     'description': 'Unix shell regex, where * matches everything.',
     'type': 'string'}},
   'required': ['pattern']}}
   ```

**Comment by keyboardAnt** [comment]
We have the same issue [here](https://github.com/minimaxir/simpleaichat/issues/40) 👋

**Comment by ahakanbaba** [comment]
> Hi, @ahakanbaba! I'm Dosu, and I'm here to help the LangChain team manage their backlog. I wanted to let you know that we are marking this issue as stale.
> 
> From what I understand, the issue you reported is about the `format_tool_to_openai_function` function in the Langchain library including unnecessary fields in the generated output. Another user, keyboardAnt, also reported a similar issue in another repository, mentioning that OpenAI's format differs from Pydantic's.
> 
> However, it seems that this issue is still unresolved. Before we proceed, we would like to confirm if this issue is still relevant to the latest version of the LangChain repository. If it is, please let us know by commenting on this issue. Otherwise, feel free to close the issue yourself, or the issue will be automatically closed in 7 days.
> 
> Thank you for your understanding and cooperation. We look forward to hearing from you soon.

As far as I can tell, yes the issue is still relevant to the latest version of langchain

**Comment by nishantkr18** [comment]
Hey. Do we have any updates on this issue?

**Comment by baskaryan** [comment]
i believe fixed in #16485, let me know if I missed something!
